### DISCUSSION TITLE
How to shard with coverage enabled?

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [vitest-dev/vitest#4755] [Q&A] [answered]
Follow up from https://github.com/vitest-dev/vitest/discussions/4752

Is there an easy way to shard with coverage enabled? Playwright for example has a merge coverage reports sub-command.

https://playwright.dev/docs/test-sharding#merge-reports-cli

**Comment by AriPerkkio** [comment, marked as answer]
Vitest does not provide any tooling or support for merging coverage reports of multiple Vitest runs. Using `--shard` option requires multiple Vitest runs.

You could use third party tools like [`nyc merge`](https://github.com/istanbuljs/nyc#combining-reports-from-multiple-runs) or [`istanbul-merge`](https://github.com/ljharb/istanbul-merge) to combine the coverage reports manually. However you'd need to make sure each `vitest --shard` run generates coverage reports with unique names so that reports from earlier runs are not overwritten. 

If you open a feature request for this I think we could implement report merging similarly as Playwright does. Vitest could also detect the `--shard` option and automatically postfix the coverage reports.

**Reply by jasonkuhrt** [reply]
I'll make an issue, thanks @AriPerkkio !

**Reply by kellyrmilligan** [reply]
was there an issue created?

**Reply by AriPerkkio** [reply]
Yes, https://github.com/vitest-dev/vitest/issues/5125

**Reply by jasonkuhrt** [reply]
Ah thanks... my bad for forgetting!

**Comment by jackonwei** [comment]
Hey, I have a question, I want to know how you do it, after the coverage report is exported, only a single test of one file is written, but the other files are 100% by default, and the overall project project coverage does not include the untested files, only the coverage of the written single test file is calculated, which is not what I want
